### PR TITLE
PageStorage: Fix data race in BlobStore

### PR DESCRIPTION
**Pull request by JaySon-Huang**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tiflash/issues/4545

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: dbms/src/Storages/Page/V3/BlobStore.cpp [hunk 1]
```diff
@@ -307,7 +303,7 @@ void BlobStore::removePosFromStats(BlobFileId blob_id, BlobFileOffset offset, si
 {
     const auto & stat = blob_stats.blobIdToStat(blob_id);
     auto lock = stat->lock();
-    stat->removePosFromStat(offset, size);
+    stat->removePosFromStat(offset, size, lock);
```
Comment (jiaqizho) [inline dbms/src/Storages/Page/V3/BlobStore.cpp:306]: if the current thread already holds a `lock_guand` from L305, Does it still need to pass the lock into the sub-methods?

Comment (jiaqizho) [reply dbms/src/Storages/Page/V3/BlobStore.cpp:306]: how about changing `stat->lock()` back to hold the `lock` from thread. then we won't pass the lock_guard many functions...

**Review by jiaqizho — APPROVED** [review]
LGTM with a suggestion: how about changing back `stat->lock()` to hold the lock from the thread. then we won't pass the lock_guard many functions.

**Review by SchrodingerZhu — APPROVED** [review]

**Comment by JaySon-Huang** [comment]
> LGTM with a suggestion: how about changing back `stat->lock()` to hold the lock from the thread. then we won't pass the lock_guard many functions.

Adding the lock_guard to the signature of the function is mainly to remind the caller should take care of the concurrent problem, and easier for reviewing

**Pull request merged**